lib/flash_flow/release/percy_client.rb

```diff
@@ -1,34 +1,56 @@
 require 'percy'
+require 'flash_flow/git'
+require 'flash_flow/mailer'
+require 'flash_flow/release/pdf_diff_generator'
 
 module FlashFlow
   module Release
     class PercyClient
 
       def initialize(config={})
-        @client = initialize_connection!(config['token'])
+        @client = initialize_connection!(config)
+        @git = ShadowGit.new(Config.configuration.git, Config.configuration.logger)
+        @mailer = FlashFlow::Mailer::Base.new(Config.configuration.smtp)
       end
 
       def find_latest_by_sha(sha)
         response = get_builds
         commit = find_commit_by_sha(response, sha)
-        build = find_build_by_commit_id(response, commit['id'])
+        find_build_by_commit_id(response, commit['id'])
+      end
+
+      def send_release_email
+        build = find_latest_by_sha(get_latest_sha)
+
+        if has_unapproved_diffs?(build)
+          @mailer.deliver!(percy_build_url: build['web-url'])
+        end
+      end
 
-        { url: build['web-url'], approved: !build['approved-at'].nil? }
+      def gen_pdf_diffs(output_file, threshold=0.0)
+        # TODO: Switch this over to Percy.get_comparisons at some point
+        PdfDiffGenerator.new.generate(get_comparisons(get_build_id), output_file, threshold)
       end
 
       private
 
-      def initialize_connection!(token)
-        if token.nil?
-          raise RuntimeError.new("Percy token must be set in your flash flow config.")
+      def get_build_id(sha=nil)
+        build = find_latest_by_sha(sha || get_latest_sha)
+        build['web-url'].split('/').last
```

Comment:
I mean I filtered only `attributes` keys in the `find_build_by_commit_id` method.
`find_latest_by_sha` can give `build['id']` for you if you want. But getting it from `build['web-url']` is also ok.